waiter/src/waiter/service_description.clj

```diff
@@ -496,7 +496,15 @@
 (defn- token->token-data
   "Retrieves the data stored against the token in the kv-store."
   [kv-store ^String token allowed-keys error-on-missing include-deleted]
-  (let [{:strs [deleted run-as-user] :as token-data} (when token (kv/fetch kv-store token))
+  (let [{:strs [deleted run-as-user] :as token-data}
+        (when token (try
+                      (kv/fetch kv-store token)
+                      (catch Exception e
+                        (let [{:keys [ex-type] :as exception-data} (ex-data e)]
+                          (if (= ex-type ::kv/invalid-zk-key)
```

Comment:
I don't like that we need to know this came from ZK, using just `kv/invalid-key` would be better as it could be thrown from any of the KV implementations.